### DISCUSSION TITLE
Set Proxy Settings in VSCode and Thunderclient for a successfull run is not working

### DISCUSSION THREAD

**Discussion by flomickl** [thunderclient/thunder-client-support#1553] [Q&A]
Hi,

I need to add some proxy settings in my setup to achieve a successful request run.
I have looked here [1] and here [2] on how to do that. I do not use the CLI. I run everything in the VSCode GUI.
But instead of getting a response, my proxy is still blocking the request.
I don't know how to set up Thunderclient so that the request is executed.

I connect to the server via remote-ssh, from which I execute the curl and thunderclient command!

From here I run the following curl command successfully in the VSCode terminal: IP is a normal single IP but changed in my example.

```
curl --noproxy "192.xxx.xxx.xxx"  -X POST   'http://192.xxx.xxx.xxx:8080/endpoint/v2/xxx/geojson'   -H 'Content-Type: application/json; charset=utf-8'   -H 'Accept: application/json, application/geo+json, application/gpx+xml, img/png; charset=utf-8'  -d '{"coordinates":[[8.681495,49.41461],[8.687872,49.420318]]}'
```

but nothing is working for thunderclient and I get a 403 Forbidden Access Denied due the proxy

My collection request json with the request looks like this

```
{
    "clientName": "Thunder Client",
    "collectionName": "test",
    "collectionId": "xxxxxxxxxxx",
    "dateExported": "xxxxxxxxxx",
    "version": "1.2",
    "folders": [],
    "requests": [
        {
            "_id": "xxxxxxxxx",
            "colId": "xxxxxxxx",
            "containerId": "",
            "name": "testing",
            "url": "v2/xxx/geojson",
            "method": "POST",
            "sortNum": 10000,
            "created": "2024-05-27T13:51:36.302Z",
            "modified": "2024-05-27T13:53:25.932Z",
            "headers": [],
            "body": {
                "type": "json",
                "raw": "{\n  \"coordinates\": [\n    [\n      8.681495,\n      49.41461\n    ],\n    [\n      8.687872,\n      49.420318\n    ]\n  ]\n}",
                "form": []
            }
        }
    ],
    "settings": {
        "options": {
            "baseUrl": "http://192.xxx.xxx.xxx:8080/endpoint/"
        }
    },
    "ref": "xxxxxxxxxxxxxxxx"
```

I added  in my settings.json

```
{
    "http.proxy": "http://proxy.url:1000",
    "http.proxyStrictSSL": false,
    "https.proxy": "http://proxy.url:1000",
    "https.proxyStrictSSL": false,
    "http.proxySupport": "override",
    "https.proxySupport": "override",
    "http.proxyOverride": "127.0.0.1, /192.xxx.xxx.xxx",
    "https.proxyOverride": "127.0.0.1, 192.xxx.xxx.xxx"
}
```
 and also added this IP in -> Thunder-client: Exclude Proxy Host List

Any ideas what to do?

[1] https://code.visualstudio.com/docs/setup/network#_proxy-server-support
[2] https://docs.thunderclient.com/features/proxy

**Comment by rangav** [comment]
try below options 
1. https://code.visualstudio.com/docs/setup/network#_proxy-server-support

2. You should add vscode application to your firewall unblock list.

**Reply by flomickl** [reply]
Thanks for the reply.
I checked option 1.

First I checked the environment variables
the environment variables on the server are set up, so thunderclient should pick up these setting, I guess

````
no_proxy=127.0.0.1,192.xxx.xxx.xxx
NO_PROXY=127.0.0.1,192.xxx.xxx.xxx
`````

If I check in the terminal via curl and verbose output, these variables are used and as it is mentioned in curl
`Uses proxy env variable no_proxy == '127.0.0.1,192.xxx.xxx.xx'

But thunderclient request is still blocked by the proxy!
So my question here to check if thunderclient is picking up these variables too:
Is there a way to do a verbose request in thunderclient, to check if these variables are used?

Second check on option 1.
I am using the remote-ssh extension to connect to the remote host, where thunderclient is running.
So I cannot start vscode via
```
code --no-proxy-server
or code --proxy-bypass-list="127.0.0.1, 192.xxx.xxx.xxx"
`````


Could this be a problem using the remote-ssh extension?

To Option 2.
I don't have access to the firewall and cannot use the CLI to test it otherwise at the moment.
Could it be really a problem with the firewall if curl is working as expected?

**Reply by rangav** [reply]
Are you paid version or free version of the extension?

**Reply by flomickl** [reply]
Currently free version due to testing and validation

**Reply by rangav** [reply]
Free version does not include support, please use stackoverflow for community support
